src/public_key.rs

```diff
@@ -51,6 +51,14 @@ impl PublicKey {
     pub fn decode_from_zbase32<I: Decodable>(encoded: I) -> Result<Self> {
         utils::decode(encoded)
     }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
```

Comment:
Do we still need this method?

It seems a little risky since we erase the information about which variant has been serialised.  Also, the `PublicKey` type already derives `Serialize`, so I don't see a use case for this `to_bytes()` method.